boa/src/string.rs

```diff
@@ -60,10 +60,13 @@ impl Inner {
         unsafe { NonNull::new_unchecked(inner) }
     }
 
-    /// Concatinate two string.
+    /// Concatinate array of strings.
     #[inline]
-    fn concat(x: &str, y: &str) -> NonNull<Inner> {
-        let total_string_size = x.len() + y.len();
+    fn concat_array(strings: &[&str]) -> NonNull<Inner> {
+        let mut total_string_size = 0;
+        for string in strings {
+            total_string_size += string.len();
+        }
```

Comment:
Maybe it would be useful to accept any slice of `&[S]` where `S: AsRef<str>` to be able to pass a `&[JsString]`, but I don't know if we even concat several `JsString` anywhere in the codebase.

Reply:
> Maybe it would be useful to accept any slice of `&[S]` where `S: AsRef<str>` to be able to pass a `&[JsString]`.

Hmmm. I'll add that, since this is exposed to the user it should also ergonomic

> but I don't know if we even concat several JsString anywhere in the codebase.

There are some in JSON parse and stringify (#1495 ) also in `RegExp.prototype.toString ( )`, `String.prototype.replaceAll`, `Date.prototype.toUTCString ( )`, and many other places just do text search `string-concatenation` and you'll see them.

Reply:
> There are some in JSON parse and stringify (#1495 ) also in `RegExp.prototype.toString ( )`, `String.prototype.replaceAll`, `Date.prototype.toUTCString ( )`, and many other places just do text search `string-concatenation` and you'll see them.

Ah yeah, we do concat more than two `JsString` on several places. Then yeah, we would avoid allocating `String`s with this change.